src/meetingsession/DefaultMeetingSession.ts

```diff
@@ -61,6 +63,8 @@ export default class DefaultMeetingSession implements MeetingSession, Destroyabl
       ),
       this._eventReporter
     );
+    this.cspViolationDetector = new DefaultCSPViolationDetector(this._logger);
+    this.cspViolationDetector.start();
```

Comment:
You’re going to get a lot of dupes…

Reply:
I tested in the demo app by appending a CSP and triggering the violation, I only saw one error was printed. Are you saying the detector will be started multiple times in one meeting session?

Reply:
Make two meetings in a single page and see what happens. If I'm reading correctly, you'll get two meeting sessions, two detectors, and thus two errors.

Reply:
My summary for this when we discussed this, I think, was: CSP violation is a property of the _page_, not the meeting.

Reply:
To get one error message per page, how about moving the implementation into static methods? When all meeting sessions stop this detector, the SDK can remove the listener.

This snippet uses the browser's `console` object. We can also keep track of `Logger` instances from multiple sessions and output an error message per unique logger.

```ts
export default class DefaultCSPViolationDetector implements CSPViolationDetector {
  private static count: number = 0;

  private static start(): void {
    if (DefaultCSPViolationDetector.count === 0) {
      document.addEventListener('securitypolicyviolation', DefaultCSPViolationDetector.listener);
    }
    DefaultCSPViolationDetector.count += 1;
  }

  private static stop(): void {
    if (DefaultCSPViolationDetector.count > 0) {
      DefaultCSPViolationDetector.count -= 1;
      if (DefaultCSPViolationDetector.count === 0) {
        document.removeEventListener('securitypolicyviolation', DefaultCSPViolationDetector.listener);
      }
    }
  }

  private static listener = (event: SecurityPolicyViolationEvent): void => {
    console.error(
      'Security Policy Violation\n' +
        `Blocked uri: ${event.blockedURI}\n` +
        `Violated directive: ${event.violatedDirective}\n` +
        `Original policy: ${event.originalPolicy}\n` +
        `Document URI: ${event.documentURI}\n` +
        `Source file: ${event.sourceFile}\n` +
        `Line No.: ${event.lineNumber}\n`
    );
  };

  private started: boolean = false;

  start(): void {
    if (!this.started) {
      this.started = true;
      DefaultCSPViolationDetector.start();
    }
  }

  stop(): void {
    if (this.start) {
      this.started = false;
      DefaultCSPViolationDetector.stop();
    }
  }
}
```